pkg/usage/usage.go

```diff
@@ -0,0 +1,267 @@
+	return u
+}
+
+// SetTrackingID Sets the GA-code for the project
+func (u *Usage) SetTrackingID(track string) *Usage {
+	u.trackID = track
+	return u
+}
+
+// SetCampaignSource : source of openebs installater like:
+// helm or operator etc. This will have to be configured
+// via ENV variable OPENEBS_IO_INSTALLER_TYPE
+func (u *Usage) SetCampaignSource(campaignSrc string) *Usage {
+	u.campaignSource = campaignSrc
+	return u
+}
+
+// SetDocumentTitle : usecase(anonymous-id)
+func (u *Usage) SetDocumentTitle(documentTitle string) *Usage {
+	u.documentTitle = documentTitle
+	return u
+}
+
+// SetApplicationName : usecase(os-type/arch, volume CASType)
+func (u *Usage) SetApplicationName(appName string) *Usage {
+	u.appName = appName
+	return u
+}
+
+// SetCampaignName : set the name of the PVC or will be empty.
+func (u *Usage) SetCampaignName(campaignName string) *Usage {
+	u.campaignName = campaignName
+	return u
+}
+
+// SetApplicationID : usecase(OpenEBS/NDM)
+func (u *Usage) SetApplicationID(appID string) *Usage {
+	u.appID = appID
+	return u
+}
+
+// SetApplicationVersion : usecase(project-version)
+func (u *Usage) SetApplicationVersion(appVersion string) *Usage {
+	u.appVersion = appVersion
+	return u
+}
+
+// SetApplicationInstallerID : usecase(k8s-version)
+func (u *Usage) SetApplicationInstallerID(appInstallerID string) *Usage {
+	u.appInstallerID = appInstallerID
+	return u
+}
+
+// SetClientID sets the anonymous user id
+func (u *Usage) SetClientID(userID string) *Usage {
+	u.clientID = userID
+	return u
+}
+
+// SetCategory sets the category of an event
+func (u *Usage) SetCategory(c string) *Usage {
+	u.category = c
+	return u
+}
+
+// SetAction sets the action of an event
+func (u *Usage) SetAction(a string) *Usage {
+	u.action = a
+	return u
+}
+
+// SetLabel sets the label for an event
+func (u *Usage) SetLabel(l string) *Usage {
+	u.label = l
+	return u
+}
+
+// SetValue sets the value for an event's label
+func (u *Usage) SetValue(v int64) *Usage {
+	u.value = v
+	return u
+}
+
+// Build is a builder method for Usage struct
+func (u *Usage) Build() *Usage {
+	// Default ApplicationID for openebs project is OpenEBS
+	v := NewVersion()
+	v.getVersion(false)
+	u.SetApplicationID(AppName).
+		SetTrackingID(GAclientID).
+		SetClientID(v.id).
+		SetCampaignSource(v.installerType)
+	// TODO: Add condition for version over-ride
+	// Case: CAS/Jiva version, etc
+	return u
+}
+
+// ApplicationBuilder Application builder is used for adding k8s&openebs environment detail
+// for non install events
+func (u *Usage) ApplicationBuilder() *Usage {
+	v := NewVersion()
+	v.getVersion(false)
+	u.SetApplicationVersion(v.openebsVersion).
+		SetApplicationName(v.k8sArch).
+		SetApplicationInstallerID(v.k8sVersion).
+		SetDataSource(v.nodeType)
+	return u
+}
+
+// SetVolumeCapacity sets the storage capacity of the volume for a volume event
+func (u *Usage) SetVolumeCapacity(volCapG string) *Usage {
+	s, _ := toGigaUnits(volCapG)
```

Comment:
error swallowed ... at least we can log IMO ...

Reply:
Its same code as maya/zfslocalpv, i think we don't want to log these errors, as they can over populate the logs and we can miss the actual business logs , so i didn't made these changes at first place . cc @vharsh

Reply:
Can take as an improvement in the next PRs.